core/src/subgraph/context/instance.rs

```diff
@@ -166,4 +168,10 @@ where
     pub(super) fn hosts(&self) -> &[Arc<T::Host>] {
         &self.hosts
     }
+
+    pub(super) fn causality_region_next_value(&mut self) -> i32 {
```

Comment:
can you elaborate a bit on this? Why do we need sequencial numbers as causality regions? 

I also question if this is the right place for this logic. If causality regions have logic to them perhaps adding it to the type would be a better place?